### ISSUE TITLE
http provider doesn't cache results

### ISSUE THREAD

**Issue by postatum** [lestrrat/go-jsref#9]
Hi.
First of all, thanks for a great project!

I've noticed a potential issue:
[Here](https://github.com/lestrrat/go-jsref/blob/master/provider/http.go#L27-L28) it says `Note that once a document is read, it WILL be cached for the duration of this object, unless you call Reset` but it doesn't actually call `hp.mp.Set()` similar to how [fs provider does](https://github.com/lestrrat/go-jsref/blob/master/provider/fs.go#L67).

I didn't test whether it actually makes requests on 2+ calls but the code looks like it will.

**Comment by lestrrat** [comment]
Yeah, it's missing a call to `Set`. And now that I look at it, I'm not sure if I like the API either.

I'll think about it, but if you are up to it, maybe you could write up a failing test case using `net/http/httptest` as the provider source? :D
